Matrix.cpp

```diff
@@ -0,0 +1,89 @@
+#include <random>
+#include <chrono>
+#include <iostream>
+
+#define MAXDIM 512
+#define RUNS 10
+
+using namespace std;
+
+random_device rd;
+mt19937 rng(rd());
+uniform_int_distribution<int> uni(0, 100);
+
+double randBetween() {
+    return uni(rng);
+}
+
+class matrix
+{
+public:
+    matrix(int dim)
+        : dim_(dim)
+    {
+        data_ = new int[dim_ * dim_];
+
+    }
+
+    inline int dim() const {
+        return dim_;
+    }
+    inline int& operator()(unsigned row, unsigned col) {
+        return data_[dim_ * row + col];
+    }
+
+    inline int operator()(unsigned row, unsigned col) const {
+        return data_[dim_ * row + col];
+    }
+
+private:
+    int dim_;
+    int* data_;
+};
```

Comment:
As this class is being used now I think you can make it be implemented on the stack using an std::array.
Unless of course it is intended to be dynamically created.